addon/components/sl-select.js

```diff
@@ -228,67 +321,48 @@ export default Ember.Component.extend( InputBased, TooltipEnabled, {
                         Ember.warn( warning, !values.length );
                     }
 
-                    return callback(
-                        multiple ?
-                        filteredContent :
-                        Ember.get( filteredContent, 'firstObject' )
-                    );
-                },
-
-                minimumResultsForSearch: this.get( 'disableSearch' ) ? -1 : 0,
-
-                query: ( query ) => {
-                    const content = this.get( 'content' ) || [];
-                    const optionLabelPath = this.get( 'optionLabelPath' );
-                    const select2 = this;
-
-                    query.callback({
-                        results: content.reduce( ( results, item ) => {
-                            const typeOfItem = Ember.typeOf( item );
-                            const text = 'object' === typeOfItem ||
-                                'instance' === typeOfItem ?
-                                Ember.get( item, optionLabelPath ) :
-                                item;
-
-                            if (
-                                text &&
-                                select2.matcher( query.term, text.toString() )
-                            ) {
-                                results.push( item );
-                            }
-
-                            return results;
-                        }, [] )
-                    });
+                    if ( filteredContent.length > 0 ) {
+                        callback( multiple ? filteredContent : filteredContent[0] );
+                    } else {
+                        callback( [] );
+                    }
```

Comment:
Really just personal sanity. If there was nothing matched for a single select, filteredContent[0] would return undefined. This way we can consistently return an a type of {Array|String}